sound/soc/sof/amd/acp.c

```diff
@@ -284,9 +284,18 @@ static irqreturn_t acp_irq_thread(int irq, void *context)
 
 	val = snd_sof_dsp_read(sdev, ACP_DSP_BAR, ACP_DSP_SW_INTR_STAT);
 	if (val & ACP_DSP_TO_HOST_IRQ) {
+		while (snd_sof_dsp_read(sdev, ACP_DSP_BAR, ACP_AXI2DAGB_SEM_0)) {
+			/* Wait until acquired HW Semaphore lock */
+			udelay(5);
+		};
```

Comment:
I don't fully get the concept of taking a hardware semaphore to deal with interrupts INSIDE an IRQ thread. Should this be part of the handler?

Reply:
We actually want to protect race condition around few shared registers in ipc thread. We aren't doing anything in handler other than waking thread also we don't want to block or delay handler hence put this semaphore in thread context.

Reply:
adding a comment along the lines above would help. The commit message hints at a very low-level IRQ issue but your reply hints at a need to protect shared registers, that's rather different.